code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -229,6 +234,22 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 	target_mob = null
 	return
 
+/mob/living/simple_animal/hostile/proc/Check_fire(target_mob)
+	if(!target_mob)
+		return FALSE
+
+	var/target_hit = FALSE
+
+	for(var/M in check_trajectory(target_mob, src, pass_flags=PASSTABLE))
+		if(ismob(M))

```

Comment:
This is an unnecessarily complicated. `for (var/mob/M` would unwind to this exact same code. So please just type the for loop.